### PR TITLE
Mapping tests for subject for H2

### PR DESCRIPTION
**Pull request by justinlittman**
…to MODS to COCINA (for H2)

## Why was this change made?
* Add tests for H2 subjects.
* Add testing for cocina-first mapping.

## How was this change tested?
Unit

## Which documentation and/or configurations were updated?
NA

### CODE REVIEW

File: spec/services/cocina/mapping/descriptive/mods/subject_h2_spec.rb [hunk 1]
```diff
@@ -0,0 +1,294 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+RSpec.describe 'MODS subject <--> cocina mappings for H2' do
```
Comment (ndushay) [inline spec/services/cocina/mapping/descriptive/mods/subject_h2_spec.rb:5]: should this read `Cocina <--> MODS mappings for H2 subjects' or even `Cocina <--> MODS mappings for FAST subjects`?

**Review by ndushay — APPROVED** [review]
LGTM;

@arcadiafalcone do you want these h2 specs:
- to keep h2 in the spec file name and description
- to be in the same directory as the other MODS specs?

**Review by ndushay — CHANGES_REQUESTED** [review]
oops - meant to request changes so Arcadia can answer

File: spec/services/cocina/mapping/descriptive/mods/subject_h2_spec.rb (hunk 1 above)
Comment (arcadiafalcone) [reply spec/services/cocina/mapping/descriptive/mods/subject_h2_spec.rb:5]: They're really Cocina --> MODS mappings at this point - they're not roundtripped back to Cocina. `Cocina --> MODS mappings for FAST subjects` is probably the most accurate.

Comment (arcadiafalcone) [reply spec/services/cocina/mapping/descriptive/mods/subject_h2_spec.rb:5]: Let's keep H2 in the file names and descriptions -- these mappings are really dependent on the structure of the H2 metadata entry form. Having them in a different directory would be preferable.

Comment (justinlittman) [reply spec/services/cocina/mapping/descriptive/mods/subject_h2_spec.rb]: OK. Moved them to an `h2` directory and used the description that you provided.

Comment (ndushay) [reply spec/services/cocina/mapping/descriptive/mods/subject_h2_spec.rb]: ugh - should it be a subdirectory of MODS?  since it's still MODS?

**Review by ndushay — APPROVED** [review]
one last question for @arcadiafalcone about whether h2 should be a subdirectory for MODS and one for @justinlittman double checking these aren't already in our old style subject specs.

Approving with those caveats.

File: spec/services/cocina/mapping/descriptive/mods/subject_h2_spec.rb (hunk 1 above)
Comment (arcadiafalcone) [reply spec/services/cocina/mapping/descriptive/mods/subject_h2_spec.rb]: It's primarily an H2>Cocina mapping though - the MODS is secondary. I don't think it should be a subdirectory of MODS.

Comment (justinlittman) [reply spec/services/cocina/mapping/descriptive/mods/subject_h2_spec.rb]: We can always change later if `H2` is no longer appropriate.

**Comment by justinlittman** [comment]
I think contributor and form are the only area with old style tests for H2.

**Pull request merged**